### ISSUE TITLE
Export-AzureVM not working

### ISSUE THREAD

**Issue by cmerat** [Azure/azure-powershell#1944]
Windows 10 Pro. Using SDK version 1.2.3 installed via the Web Platform Installer (shows as 1.2.2 in the WPI).

```
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   1.2.3      Azure                               {Add-AzureAccount, Add-AzureApplicationGatewaySslCertifica...
```

Getting the following error when attempting to Export-AzureVM

```
PS C:\WINDOWS\system32> Export-AzureVM -ServiceName XXX -Name XXX -Path c:\temp\VM.xml
Export-AzureVM : Could not read properties for virtual machine: XXX. It may still be provisioning.
At line:1 char:1
+ Export-AzureVM -ServiceName game2010 -Name game2010 -Path c:\temp\Azu ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Export-AzureVM], ApplicationException
    + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.ExportAzureVMCommand
```

Using an alternate computer with Windows 8.1 and the 0.9.3 SDK works as expected.

**Comment by markcowl** [comment]
1.2.3 is the version of the Azure Module.  1.2.2 is the version fo the SDK (and the AzureRM module).

Adding the appropriate folks from the service team to take a look.

@singhkay can you take a look

**Comment by singhkay** [comment]
@cmerat There haven't been any changes to this cmdlet in a few years so everything should be the same. Was the VM in this case up and running fine?

**Comment by cmerat** [comment]
@singhkay Yes the VM was up and running. In fact, the older cmdlets worked (from an alternate workstation).

**Comment by singhkay** [comment]
@cmerat Repros for me as well. I'll file an investigation on the dev team.

**Comment by craiglandis** [comment]
It reproduces for me also. VMs I tested on are all in ReadyRole, so I don't know why the error says "It may still be provisioning" (full error is "export-azurevm : Could not read properties for virtual machine: <VM name>. It may still be provisioning"

**Comment by BorisB2015** [comment]
Hit this just now and have an active case on it. Linux VM in my case (not sure if ConsoleScreenshotBlobUri unique to Linux VMs or not).

Full stack on this:

- Could not read properties for virtual machine: blah. It may still be provisioning.
- There was an error reflecting type 'Microsoft.WindowsAzure.Commands.ServiceManagement.Model.PersistentVM'.
- There was an error reflecting property 'DebugSettings'.
- There was an error reflecting type 'Microsoft.WindowsAzure.Commands.ServiceManagement.Model.DebugSettings'.
- Cannot serialize member 'Microsoft.WindowsAzure.Commands.ServiceManagement.Model.DebugSettings.ConsoleScreenshotBlobUri' of type 'System.U
- details.
- System.Uri cannot be serialized because it does not have a parameterless constructor.

**Comment by BorisB2015** [comment]
cc: @singhkay 

Repored on version: Azure 1.2.3, ASM mode.

**Comment by rginjupalli** [comment]
Till issue is fixed permanently, one of the temporary workaround is to use Get-AzureVM

(Get-AzureVM -ServiceName foo -Name bar | ConvertTo-Xml).Save("c:\output-file.xml")

**Comment by retrobits** [comment]
@rginjupalli - this does not seem to be an acceptable work around, as the output is different
I have several VMs (Ubuntu) and all give me the same error about "reading properties"

**Comment by rginjupalli** [comment]
@retrobits - Capturing the VM configuration using Get-AzureVM and then converting the result to xml file (Isn't Export-AzureVM does the same thing)

**Comment by Aendapally** [comment]
Export-AzureVM was failing for me so I have got the config XML file by Get-AzureVM -ServiceName foo -Name bar | ConvertTo-Xml).Save("c:\output-file.xml") and after that ----
Import-AzureVM is not working.. any suggestions on this ?

PS C:\WINDOWS\system32> Import-AzureVM -Path c:\temp\VM.xml
Import-AzureVM : There was an error reflecting type
'Microsoft.WindowsAzure.Commands.ServiceManagement.Model.PersistentVM'.
At line:1 char:1
+ Import-AzureVM -Path c:\temp\Azure\game2013-Raw.xml
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Import-AzureVM], InvalidOperationException
    + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.ImportAzureVMCommand

**Comment by Aendapally** [comment]
PS C:\Users\akhileshwar\Desktop> Import-AzureVM -Path 'C:\Users\akhileshwar\Desktop\HAYSTATTC02.xml' | New-AzureVM -ServiceName 'Activate-Test-Public' -Location 'East US'
Import-AzureVM : There was an error reflecting type 'Microsoft.WindowsAzure.Commands.ServiceManagement.Model.PersistentVM'.
At line:1 char:1
+ Import-AzureVM -Path 'C:\Users\akhileshwar\Desktop\HAYSTATTC02.xml' | ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Import-AzureVM], InvalidOperationException
    + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.ImportAzureVMCommand

**Comment by singhkay** [comment]
This was fixed in #2045. Please give it a try and let us know.

**Comment by singhkay** [comment]
Reopening since this is still broken. @hyonholee Can you take a look?
